### PR TITLE
Rename VS Code extension to rust-analyzer

### PR DESCRIPTION
**Pull request by matklad**
I want to merge this before release on Monday, such that we can give heads up on twitter

### CODE REVIEW

**Comment by gilescope** [comment]
It might be quite nice to rename the ra_lsp_server to rust_analyser_server also so it's a little clearer what that process is.

**Comment by bjorn3** [comment]
Or `rust_analyzer_lsp_server`.

**Comment by matklad** [comment]
I guess just `rust-analyzer` will be good. In theory, we are much more than an LSP server. In practice, we are this special case basically :D 

I'd rather do it separately though, changing two things at a time is worse than doing it in two steps

File: editors/code/package.json [hunk 1]
```diff
@@ -1,10 +1,11 @@
 {
-    "name": "ra-lsp",
-    "displayName": "ra-lsp",
+    "name": "rust-analyzer",
+    "displayName": "rust-analyzer",
```
Comment (Matelasse) [inline editors/code/package.json]: Maybe capital case this? Most extensions are capital cased

Comment (matklad) [reply editors/code/package.json]: Most, but not all, so I think it's better to stick to the "official" spelling of the project name: `rust-analyzer`. Should have chosen something single-worded and captializable instead, but too late to change now :)

**Pull request merged**